tests/utils_tests/test_datastructures.py

```diff
@@ -55,6 +55,10 @@ def test_len(self):
         s.add(2)
         self.assertEqual(len(s), 2)
 
+    def test_reversed(self):
+        s = OrderedSet([1, 2, 3])
+        self.assertEqual(list(reversed(s)), [3, 2, 1])
```

Comment:
Before checking the return value's contents, you should check that its type is `OrderedSet`.

Reply:
Sorry, the return value should be an iterator, not an `OrderedSet`. It might be good to check that.

Reply:
`reversed(s)` evaluates to something like `<dict_reversekeyiterator object at 0x7fc4ea8d56d0>`, just like `iter(s)` evaluates to  `<dict_keyiterator object at 0x7fc4ea7554f0>`: 

```python
from django.utils.datastructures import OrderedSet
s = OrderedSet([1, 2, 3])
print(reversed(s)) # <dict_reversekeyiterator object at 0x7fc4ea8d56d0>
print(iter(s))     # <dict_keyiterator object at 0x7fc4ea8d56d0>
```

That's OK, right?

Reply:
Yes, that's okay. That's a kind of iterator. In your test, one way to do it would be to check that the type of the return value is the same as `type(reversed({}))`, maybe with a comment that you're doing that to check that it's _some_ iterator, rather than that particular one. If you wanted to be fancier, there's probably a way to do it using `collections.abc`: https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterator

Reply:
Well, I considered checking that, but I discarded it because that's actually just testing the native python implementation `reversed(d: dict)`.

I also considered testing `OrderedSet`s with miscellaneous data types (like user-defined class instances) rather than just integers. But then again, that would be testing python's set checking, be it through `__id__` (for unhashable custom objects) or `__hash__` and `__eq__`. I figured that's why the previous tests don't bother with that.

Do you think it's adequate to extend not only mine, but also the previous tests?

Reply:
> Well, I considered checking that, but I discarded it because that's actually just testing the native python implementation 

No, it would be useful because it's checking what `OrderedSet. __reversed__()` returns. It's just that the answer is in terms of a Python dict.

Reply:
> I also considered testing OrderedSets with miscellaneous data types (like user-defined class instances) rather than just integers. 

You don't need to check with other data types, or modify other tests. Using integers is fine.